### PR TITLE
Fix build on OCaml 4.06 and clean up warnings

### PR DESCRIPTION
**Pull request by anmonteiro**
You really need to start running `dune build` without the release profile to catch all these warnings..

### CODE REVIEW

File: src/bucklescript/output_bucklescript_serializer.re [hunk 1]
```diff
@@ -143,6 +141,19 @@ let is_recursive = input_objects => {
   List.length(input_objects) > 1;
 };
 
+let filter_map = f => {
```
Comment (anmonteiro) [inline src/bucklescript/output_bucklescript_serializer.re:144]: copied from OCaml stdlib, not available in 4.06

File: src/native/graphql_ppx.re [hunk 2]
```diff
@@ -151,18 +151,10 @@ let rewrite_query = (~schema=?, ~loc, ~delim, ~query, ()) => {
         warnings
         |> List.iter(((loc, message)) => {
              let loc = conv_loc(loc);
-             let loc_as_ghost = {...loc, loc_ghost: true};
-             Location.print_alert(
-               loc,
+             Location.report_error(
```
Comment (anmonteiro) [inline src/native/graphql_ppx.re:154]: alerts are only available since OCaml 4.08

File: src/bucklescript/graphql_ppx.re [hunk 3]
```diff
@@ -414,18 +413,10 @@ let rewrite_query =
         warnings
         |> List.iter(((loc, message)) => {
              let loc = conv_loc(loc);
-             let loc_as_ghost = {...loc, loc_ghost: true};
-             Location.print_alert(
-               loc,
+             Location.report_error(
```
Comment (anmonteiro) [inline src/bucklescript/graphql_ppx.re:416]: alerts are only available since OCaml 4.08

**Comment by anmonteiro** [comment]
I realize all the warning fixes may make reviewing this a little hard. I pinpointed the actual changes that make it compile on OCaml 4.06 in GH comments.

File: src/native/graphql_ppx.re (hunk 2 above)
Comment (anmonteiro) [reply src/native/graphql_ppx.re:154]: OK, hold on. Now `report_error` isn't available anymore in 4.10

**Comment by anmonteiro** [comment]
k, fixed in 82a1162

**Comment by jfrolich** [comment]
LGTM 🎉

**Pull request merged**